test/net/sourceforge/kolmafia/utilities/StringUtilitiesTest.java

```diff
@@ -13,8 +15,292 @@ void isVowel() {
+    "'123\n4567890123456789', '123\n4\n56789\n01234\n56789\n'",
+    "'12 45', '12\n45'",
+    "'1 3 5 7 9 1', '1 3 5\n7 9\n1'",
+    "'1 2  \n  4 5\n 6 \n', '1 2\n4 5\n6'",
+    "'1234 67', '1234\n67'"
+  })
+  public void exerciseTextWrapWithShortWrapLength(String input, String expected) {
+    assertEquals(expected, StringUtilities.basicTextWrap(input, 5));
+  }
+
+  /*
+  This builds a string of the specified length.  It uses the repeating pattern "1234567891" to
+  make it easier for humans to debug.  The modulo 10 is just to get a single digit as a string;
+   */
+  private String buildTestString(int length) {
+    StringBuilder retVal = new StringBuilder();
+    for (int i = 1; i <= length; i++) {
+      retVal.append(i % 10);
+    }
+    return retVal.toString();
+  }
+  /*
+  This inserts a break into a string
+   */
+  private String manuallyInsertBreak(String input, int breakPos) {
+    if (breakPos >= input.length()) {
+      return input;
+    } else {
+      return input.subSequence(0, breakPos) + BREAK + input.subSequence(breakPos, input.length());
+    }
+  }
+  // End tests for basicTestWrap
+
+  @ParameterizedTest
+  @CsvSource({
+    "'not an id', '-1'",
+    "'[not an id]', '-1'",
+    "'123','-1'",
+    "'[123', '-1'",
+    "'123[', '-1'",
+    "'123', '-1'",
+    "'[1337]2468 googles', '1337'",
+    "'[123]unreal item', '123'"
+  })
+  public void itShouldExerciseGetBracketedID(String name, String id) {
+    assertEquals(Integer.parseInt(id), StringUtilities.getBracketedId(name));
+  }
+
+  @ParameterizedTest
+  @CsvSource({
+    "'not an id', 'not an id'",
+    "'[not an id]', '[not an id]'",
+    "'[123', '[123'",
+    "'123]', '123]'",
+    "'123', '123'",
+    "'[123]unreal item', 'unreal item'",
+    "'[1337]2468 goggles', '2468 goggles'"
+  })
+  public void itShouldExerciseRemoveBracketedID(String name, String id) {
+    assertEquals(id, StringUtilities.removeBracketedId(name));
+  }
+
+  @Test
+  public void itShouldTokenizeStringsWithDefaults() {
+    // simple comma separated list
+    String testString = "folder (cyan), folder (magenta), folder (yellow)";
+    List<String> expected = new ArrayList<>();
+    expected.add("folder (cyan)");
+    expected.add("folder (magenta)");
+    expected.add("folder (yellow)");
+    try {
+      List<String> result = StringUtilities.tokenizeString(testString);
+      assertEquals(expected, result, "Lists are not the same.");
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void itShouldHandleGoodAndBadEscape() {
+    String testString = "a,\tb";
+    List<String> expected = new ArrayList<>();
+    expected.add("a");
+    expected.add("\tb");
+    try {
+      List<String> result = StringUtilities.tokenizeString(testString, ',', '\\', false);
+      assertEquals(expected, result, "Lists are not the same.");
+    } catch (Exception e) {
+      fail(e.getMessage());
+    }
+    char[] test = new char[4];
```

Comment:
Does `"a,\\g"` not work?

Also maybe split this into three tests? They don't seem to rely on each other.

Reply:
"a.\\g" did not work because I am explicitly trying to make a string that has invalid escape sequences.  The complier wouldn't let me.  I will look at breaking things up.

Reply:
Needs to be `a,\\g`, because `\g` is an invalid escape. To make a literal backslash in a string you have to double the character.

As far as `tokenizeString` is concerned, escaping is only relevant for commas: it's used for item names with commas in them (e.g. "tiny plastic Hank North, Photojournalist")

Reply:
testString = "a,\\\\g"; is what my IDE substitutes when I try and make a string a comma slash slash g.  The test then fails whereas building the character array, passes.  This test was driven by a desire to get 100% coverage of tokenizeString.  This is the first attempt that succeeded in covering all the branches.  tokenizeString accepts both a separator and an escape character.  The focus on command and slash is just me focusing on coverage.  Perhaps different separators and escapes need to be tried to make things clearer?